### PR TITLE
python311Packages.elevenlabs: init at 1.2.2

### PR DESCRIPTION
**Pull request by ivankovnatsky**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/elevenlabs/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{ buildPythonPackage
+, fetchPypi
+, lib
+, python3
```
Comment (fabaff) [inline pkgs/development/python-modules/elevenlabs/default.nix:4]:
```suggestion
```

File: pkgs/development/python-modules/elevenlabs/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-69AoablWAriVaHTdcnmBu0mtFrmjwvWQEZPYOCE2lKo=";
+  };
+
+  nativeBuildInputs = [
+    python3.pkgs.poetry-core
+  ];
```
Comment (fabaff) [inline pkgs/development/python-modules/elevenlabs/default.nix:19]:
```suggestion
  build-system = [
    poetry-core
  ];
```

File: pkgs/development/python-modules/elevenlabs/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  propagatedBuildInputs = with python3.pkgs; [
+    httpx
```
Comment (fabaff) [inline pkgs/development/python-modules/elevenlabs/default.nix:22]:
```suggestion
  dependencies = [
    httpx
```

File: pkgs/development/python-modules/elevenlabs/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    ipython
+    pydantic
+    requests
+    typing-extensions
+    websockets
+  ];
+
```
Comment (fabaff) [inline pkgs/development/python-modules/elevenlabs/default.nix:29]:
```suggestion
  ];

  pythonImportsCheck = [
    "elevenlabs"
  ];
```

**Review by fabaff — COMMENTED** [review]
The upstream repo contains [tests](https://github.com/elevenlabs/elevenlabs-python/tree/main/tests). Could you please give it a try with using the source from GitHub (`fetchFromGitHub`) and `pytestCheckHook`.

**Comment by ivankovnatsky** [comment]
With the latest commit and tests upstream tests enabled I'm getting:

```console
            timeout=request_options.get("timeout_in_seconds")
            if request_options is not None and request_options.get("timeout_in_seconds") is not None
            else self._client_wrapper.get_timeout(),
            retries=0,
            max_retries=request_options.get("max_retries") if request_options is not None else 0,  # type: ignore
        ) as _response:
            if 200 <= _response.status_code < 300:
                for _chunk in _response.iter_bytes():
                    yield _chunk
                return
            _response.read()
            if _response.status_code == 422:
                raise UnprocessableEntityError(
                    typing.cast(HttpValidationError, construct_type(type_=HttpValidationError, object_=_response.json()))  # type: ignore
                )
            try:
                _response_json = _response.json()
            except JSONDecodeError:
                raise ApiError(status_code=_response.status_code, body=_response.text)
>           raise ApiError(status_code=_response.status_code, body=_response_json)
E           elevenlabs.core.api_error.ApiError: status_code: 401, body: {'detail': {'status': 'needs_authorization', 'message': 'Neither authorization header >

/nix/store/gaz70dppc499drvzzh554bdvzklfsjig-python3.12-elevenlabs-1.2.2/lib/python3.12/site-packages/elevenlabs/voice_generation/client.py:179: ApiError
=============================== warnings summary ===============================
../../../../nix/store/ggjx693a2vl9sha4q1qqy50m6j0k6mm9-python3.12-pytest-8.1.1/lib/python3.12/site-packages/_pytest/config/__init__.py:1439
  /nix/store/ggjx693a2vl9sha4q1qqy50m6j0k6mm9-python3.12-pytest-8.1.1/lib/python3.12/site-packages/_pytest/config/__init__.py:1439: PytestConfigWarning: Unkno>
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_client.py::test_generate_stream - ValueError: mpv not found, necessary to stream audio. On mac you can instal...
FAILED tests/test_history.py::test_history - elevenlabs.core.api_error.ApiError: status_code: 401, body: {'detail': {'st...
FAILED tests/test_history.py::test_history_item_delete - elevenlabs.core.api_error.ApiError: status_code: 401, body: {'detail': {'st...
FAILED tests/test_model.py::test_model - elevenlabs.core.api_error.ApiError: status_code: 404, body: {'detail': {'st...
FAILED tests/test_voice.py::test_voice_from_id - elevenlabs.core.api_error.ApiError: status_code: 401, body: {'detail': {'st...
FAILED tests/test_voice.py::test_voice_clone - elevenlabs.core.api_error.ApiError: status_code: 401, body: {'detail': {'st...
FAILED tests/test_voice.py::test_voice_design - elevenlabs.core.api_error.ApiError: status_code: 401, body: {'detail': {'st...
============== 7 failed, 4 passed, 1 warning in 65.66s (0:01:05) ===============
/nix/store/g224xn9nwzf16jr94f1jx4jsnlazkl2a-stdenv-darwin/setup: line 1588: pop_var_context: head of shell_variables not a function context
```

running: `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`, weirdly `wip` or `wip --staged` would not pick up my changes, but that's another story.

**Comment by ivankovnatsky** [comment]
With the current change:

```console
❯ nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
remote: Enumerating objects: 725, done.
remote: Counting objects: 100% (668/668), done.
remote: Compressing objects: 100% (319/319), done.
remote: Total 725 (delta 432), reused 492 (delta 316), pack-reused 57
Receiving objects: 100% (725/725), 948.10 KiB | 3.56 MiB/s, done.
Resolving deltas: 100% (437/437), completed with 73 local objects.
From https://github.com/NixOS/nixpkgs
   ad86d10f138d..2999c9360467  master     -> refs/nixpkgs-review/0
$ git worktree add /Users/ivan/.cache/nixpkgs-review/rev-6d6b8927782349eee5cdb047eb194f92a0da1c01/nixpkgs 2999c9360467c46604a9f4f1cf0a2fd46c62c9c2
Preparing worktree (detached HEAD 2999c9360467)
Updating files: 100% (41134/41134), done.
HEAD is now at 2999c9360467 Merge pull request #313265 from r-ryantm/auto-update/screenly-cli
$ nix-env --extra-experimental-features no-url-literals --option system aarch64-darwin -f <nixpkgs> --nix-path nixpkgs=/Users/ivan/.cache/nixpkgs-review/rev-6d6b8927782349eee5cdb047eb194f92a0da1c01/nixpkgs nixpkgs-overlays=/tmp/tmpy9xhitl4 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git merge --no-commit --no-ff 6d6b8927782349eee5cdb047eb194f92a0da1c01
Auto-merging pkgs/top-level/python-packages.nix
Automatic merge went well; stopped before committing as requested
$ nix-env --extra-experimental-features no-url-literals --option system aarch64-darwin -f <nixpkgs> --nix-path nixpkgs=/Users/ivan/.cache/nixpkgs-review/rev-6d6b8927782349eee5cdb047eb194f92a0da1c01/nixpkgs nixpkgs-overlays=/tmp/tmpy9xhitl4 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
2 packages added:
python311Packages.elevenlabs (init at 1.2.2) python312Packages.elevenlabs (init at 1.2.2)

$ nix build --nix-path nixpkgs=/Users/ivan/.cache/nixpkgs-review/rev-6d6b8927782349eee5cdb047eb194f92a0da1c01/nixpkgs nixpkgs-overlays=/tmp/tmpy9xhitl4 --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation -f /Users/ivan/.cache/nixpkgs-review/rev-6d6b8927782349eee5cdb047eb194f92a0da1c01/build.nix
4 packages built:
python311Packages.elevenlabs python311Packages.elevenlabs.dist python312Packages.elevenlabs python312Packages.elevenlabs.dist

$ /nix/store/wmim35b9qchk098r3hqd7lyfi621z27i-nix-2.18.1/bin/nix-shell /Users/ivan/.cache/nixpkgs-review/rev-6d6b8927782349eee5cdb047eb194f92a0da1c01/shell.nix --nix-path nixpkgs=/Users/ivan/.cache/nixpkgs-review/rev-6d6b8927782349eee5cdb047eb194f92a0da1c01/nixpkgs nixpkgs-overlays=/tmp/tmpy9xhitl4

[nix-shell:~/.cache/nixpkgs-review/rev-6d6b8927782349eee5cdb047eb194f92a0da1c01]$ 
```

**Comment by wegank** [comment]
Done in https://github.com/NixOS/nixpkgs/commit/d9910c17aab10b5b914d637340f964ea2e7aafd4.

**Pull request closed without merging**